patroni/dcs/kubernetes.py

```diff
@@ -698,8 +698,9 @@ def __init__(self, config):
         self._namespace = config.get('namespace') or 'default'
         self._role_label = config.get('role_label', 'role')
         self._ca_certs = os.environ.get('PATRONI_KUBERNETES_CACERT', config.get('cacert')) or SERVICE_CERT_FILENAME
-        config['namespace'] = ''
-        super(Kubernetes, self).__init__(config)
+        config_copy = deepcopy(config)
+        config_copy['namespace'] = ''
+        super(Kubernetes, self).__init__(config_copy)
```

Comment:
Since we don't need 2.7 compatibility anymore it could be rewritten as:
```suggestion
        super(Kubernetes, self).__init__({**config, 'namespace': ''})
```